server/preprocessing/other-scripts/openaire.R

```diff
@@ -101,6 +101,29 @@ get_return_values <- function(all_artifacts){
   return (ret_val)
 }
 
+preprocess_data <- function(all_artifacts){
+  all_artifacts$subject <- unlist(lapply(all_artifacts$subject, function(x) {gsub("\\[[A-Za-z \\.-]+\\]", "", x)})) # removes [ INFO.INFO-MA ] Computer Science [cs]/Multiagent Systems [cs.MA]
+  all_artifacts$subject <- unlist(lapply(all_artifacts$subject, function(x) {gsub(" ?/", ";", x)}))
+  all_artifacts$subject <- unlist(lapply(all_artifacts$subject, function(x) {gsub("\\:.*\\:\\:", "", x)})) # keeps only last part after :: ":Enginyeria de la telecomunicació::Processament del senyal::Reconeixement de formes [Àrees temàtiques de la UPC]"
+  all_artifacts$subject <- unlist(lapply(all_artifacts$subject, function(x) {gsub("Q[A-Z]?\\d{1,2}.*\\. ?", "", x)})) # keeps only last part, here Computer science "QA75 Electronic computers. Computer science; számítástechnika, számítógéptudomány"
+  all_artifacts$subject <- unlist(lapply(all_artifacts$subject, function(x) {gsub("QA\\d{1,2}", "", x)}))
+  all_artifacts$subject <- unlist(lapply(all_artifacts$subject, function(x) {gsub("info:eu-repo;classification;ddc;\\d+", "", x)})) # ddc classification
+  all_artifacts$subject <- unlist(lapply(all_artifacts$subject, function(x) {gsub("info:eu-repo;classification;acm;\\w+.\\d+.\\d+ ", "", x)})) # acm classification
+  all_artifacts$subject <- unlist(lapply(all_artifacts$subject, function(x) {gsub("ddc:\\d+", "", x)}))
+  all_artifacts$subject <- unlist(lapply(all_artifacts$subject, function(x) {gsub("jel:[A-Z]+\\d+", "", x)}))
+  all_artifacts$subject <- unlist(lapply(all_artifacts$subject, function(x) {gsub("[A-Z]{2}\\d+-\\d+", "", x)})) # e.g. "TA1-2040"
+  all_artifacts$subject <- unlist(lapply(all_artifacts$subject, function(x) {gsub("\\w*\\:\\:", "", x)})) # keeps only last part of e.g. Physics::Optics
+  all_artifacts$subject <- unlist(lapply(all_artifacts$subject, function(x) {gsub("\\w*\\ - ", "", x)})) # keeps only last part of e.g. Physics - Atomic Physics
```

Comment:
In case of `High energy physics - lattice` it keeps `High energy lattice`. See funder="RCUK", grant ID="ST/K000411/1"

Reply:
Thanks, will add some further improvements. High energy lattice sounds like the new superfood.